### PR TITLE
Support function native arguments generation

### PR DESCRIPTION
**Pull request by SemenchenkoVitaliy**
/cc @lundibundi @belochub @nechaido

### CODE REVIEW

File: lib/introspector.js [hunk 1]
```diff
@@ -252,6 +254,18 @@ const parseFunction = (fn, lines) => {
     sig.comments = parseRest(comments);
   }
 
+  if (typeof fn === 'function') {
+    const nativeParamRegex = /\([\w\s,.]*\)/;
```
Comment (lundibundi) [inline lib/introspector.js]: Would this correctly handle default params in a function and destructuring?

File: test/example.js [hunk 2]
```diff
@@ -96,6 +96,8 @@ const noTitleFunction = (str, num) => str + num;
 // Async functions are supported
 const asyncFunc = async () => {};
 
+const undocumentedArgumentFunction = (arg1, arg2, ...args) => {};
+
```
Comment (lundibundi) [inline test/example.js]: Could you also add
```js
const undocumentedDestructureFn = (arg1, { a, b = 3, c = 'hello, world' }) => {};
const undocumentedDefaultFn = (arg1, arg2 = 'hello, world') => {}; 
```

File: lib/generator.js [hunk 3]
```diff
@@ -224,7 +224,8 @@ const generateRest = (comments, customTypes, usedTypes) => {
   for (const comment of comments) {
     let line;
     if (comment.name === 'Example' || comment.name === 'Result') {
-      const code = comment.comment.trimLeft();
+      let code = comment.comment.trimLeft();
+      if (!code.endsWith(';')) code += ';';
```
Comment (lundibundi) [inline lib/generator.js]: Would this correctly work for
```js
Example:
callAsyncFunction((a, b) => {
  if (a) {  
    console.log(b)
  }
})
```
?

Comment (SemenchenkoVitaliy) [reply lib/generator.js]: Yes, but you'll need to add `;` after `console.log(b)`.

Comment (lundibundi) [reply lib/generator.js]: Could you add a test for that, please?

**Comment by lundibundi** [comment]
@SemenchenkoVitaliy could you please squash the commits, there seem to be conflicts during `git rebase --autosquash origin/master`.

**Comment by lundibundi** [comment]
Landed in https://github.com/metarhia/metadoc/commit/201b03ad5c8cacb510767b112729c8be3dbafe68, https://github.com/metarhia/metadoc/commit/778c5c73913838d487802ab7bfae2493fe1112e0 and https://github.com/metarhia/metadoc/commit/e9c3bbcb1d8e54e32167b02f9b191bfc084e979f.

**Pull request closed without merging**